### PR TITLE
Remove six as a dependency in core

### PR DESCRIPTION
**Pull request by lmazuel**
Came in a discussion with @annatisch , and we feel it's probably time, that it wouldn't break anyone, and thee few if any would have done wrong things in the first place.

### CODE REVIEW

File: sdk/core/azure-core/setup.py [hunk 1]
```diff
@@ -70,7 +70,6 @@
     python_requires=">=3.8",
     install_requires=[
         "requests>=2.21.0",
-        "six>=1.11.0",
```
Comment (xiangyan99) [inline sdk/core/azure-core/setup.py:73]: Add it in the changelog?

Comment (lmazuel) [reply sdk/core/azure-core/setup.py:73]: Yes of course, was curious to see what CI would do, and it was a one click PR from GH UI :p

**Review by xiangyan99 — APPROVED** [review]

**Review by kashifkhan — APPROVED** [review]

**Pull request closed without merging**